### PR TITLE
Fix: List Widget crashes on null empty object in data

### LINKED ISSUE #4346: [Bug] TypeError: Cannot convert undefined or null to object
Sentry Issue: [APPSMITH-RP](https://sentry.io/organizations/appsmith/issues/2384676115/?referrer=github_integration)

```
TypeError: Cannot convert undefined or null to object
  at Function.keys (<anonymous>)
  at additionalAutoComplete (widgets/ListWidget/ListPropertyPaneConfig.ts:75:25)
  at c (pages/Editor/PropertyPane/PropertyControl.tsx:310:30)
...
(45 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by jsartisan**
This PR fixes the crash happening due to an empty data objects passed to the list widget.

Fixes #4346 

- Bugfix

## How Has This Been Tested?
- Manually tested

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/list-widget-crash-issue 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 51.55 **(0.02)** | 32.47 **(0.03)** | 29.63 **(0.06)** | 52.04 **(0.01)**
 :green_circle: | app/client/src/components/designSystems/appsmith/ContainerComponent.tsx | 92.86 **(1.56)** | 55.56 **(-1.58)** | 100 **(0)** | 92.86 **(1.56)**
 :red_circle: | app/client/src/components/designSystems/appsmith/PositionedContainer.tsx | 100 **(0)** | 78.57 **(-1.43)** | 100 **(0)** | 100 **(0)**
 :red_circle: | app/client/src/mockResponses/WidgetConfigResponse.tsx | 13.92 **(-0.37)** | 0 **(0)** | 0 **(0)** | 14.47 **(-0.39)**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/index.tsx | 82.83 **(1.01)** | 71.58 **(1.69)** | 68 **(0)** | 83.7 **(1.09)**
 :red_circle: | app/client/src/utils/hooks/useClickOpenPropPane.tsx | 80.77 **(-15.66)** | 51.35 **(-35.32)** | 100 **(0)** | 79.59 **(-16.71)**
 :green_circle: | app/client/src/widgets/ListWidget/ListComponent.tsx | 76.47 **(7.24)** | 50 **(-25)** | 42.86 **(17.86)** | 76.47 **(7.24)**
 :red_circle: | app/client/src/widgets/ListWidget/ListPropertyPaneConfig.ts | 40 **(-26.67)** | 0 **(-100)** | 0 **(0)** | 40 **(-26.67)**
 :green_circle: | app/client/src/widgets/ListWidget/ListWidget.tsx | 38 **(0.05)** | 22.83 **(2.14)** | 26.47 **(-0.8)** | 37.1 **(0.08)**</details>

### CODE REVIEW

**Review by marks0351 — DISMISSED** [review]

File: app/client/src/utils/hooks/useClickOpenPropPane.tsx [hunk 1]
```diff
@@ -29,22 +70,33 @@ export const useClickOpenPropPane = () => {
   const isDragging = useSelector(
     (state: AppState) => state.ui.widgetDragResize.isDragging,
   );
+
+  const focusedWidget = getParentToOpenIfAny(focusedWidgetId, widgets);
```
Comment (marks0351) [inline app/client/src/utils/hooks/useClickOpenPropPane.tsx:74]: can we rename focusedWidget to something more relevant?

File: app/client/src/widgets/BaseWidget.tsx [hunk 2]
```diff
@@ -243,6 +243,9 @@ abstract class BaseWidget<
     const style = this.getPositionStyle();
     return (
       <PositionedContainer
+        focused={this.props.focused}
+        resizeDisabled={this.props.resizeDisabled}
+        selected={this.props.selected}
```
Comment (marks0351) [inline app/client/src/widgets/BaseWidget.tsx:248]: is there a way to handle selected based styles within the ListComponent, I am assuming this is the style that we add when a list widget item is clicked?

File: app/client/src/widgets/ListWidget/ListComponent.tsx [hunk 3]
```diff
@@ -64,4 +66,19 @@ export const ListComponentEmpty = styled.div`
...
 `;
 
+export const ListComponentLoading = styled.div<{
+  backgroundColor?: string;
+}>`
+  height: 100%;
+  width: 100%;
+  position: relative;
+  overflow: hidden;
+  border: ${(props) => getBorderCSSShorthand(props.theme.borders[2])};
```
Comment (marks0351) [inline app/client/src/widgets/ListWidget/ListComponent.tsx:76]: can we use box-shadow instead of border to help save width?

**Review by marks0351 — CHANGES_REQUESTED** [review]

File: app/client/src/widgets/BaseWidget.tsx (hunk 2 above)
Comment (jsartisan) [reply app/client/src/widgets/BaseWidget.tsx:248]: @marks0351 No actually. All the styling is done in the individual widgets. ListWidget can only pass/modify props from top.

**Review by marks0351 — DISMISSED** [review]

**Review by marks0351 — APPROVED** [review]

**Pull request merged**